Automattic/wp-calypso
# Nav sidebar: Update styling to match latest mockup

#### Changes proposed in this Pull Request

![Jul-16-2020 09-56-12](https://user-images.githubusercontent.com/1500769/87602178-a5b79280-c74a-11ea-81f7-4f9de88079af.gif)

* Colours match mockup
* Add a heading to the list
* Add ellipses to long titles and show tooltip on hover
* Change animation (sidebar still slides out, but (W) icon no longer shrinks)
* Remove the "View all pages/posts" button, which has been removed from the mockup
* "Add new page/post" button sticks to the bottom of the sidebar when the list is long
* Reduce sidebar animation time to bring it inline with the core `<Modal>` animation time.
* Add a consistent focus ring to all elements in the sidebar

Worked off these mockups: pbAok1-185-p2#comment-2345

The site title and url seen in the mockup are still to come #43973 

There's quite a bit of code change, but each commit should be fairly independent if that helps with reviewing.

#### Testing instructions



* Clone this branch and run Calypso locally
* Sandbox `public-api.wordpress.com`, `widgets.wp.com`, and your favourite test site
* Add the `nav-sidebar-enabled` sticker to your test site _or_ define `WPCOM_BLOCK_EDITOR_SIDEBAR` in your sandbox
* Apply the `D46330-code` patch to your sandbox
* Apply the `811877` build of `wpcom-block-editor` to your sandbox (see the "automatic" instructions here: PCYsg-l4k-p2)
* Fire up the editor on your test site and try out the sidebar!
* Check that things mostly match the mockup (note that the site title and url aren't included in this PR)
* The list should scroll when there's not enough space, and the "add new page" button should stick to the bottom of the screen when scrolling

Fixes #43968 #43975
